Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Calibration/CalibrationDataHelper.cs

```diff
@@ -0,0 +1,250 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Extensions.Experimental.SpectatorView
+{
+    internal class CalibrationDataHelper
+    {
+        private const string ChessboardImageDirectoryName = "CHESS";
+        private const string ChessboardDetectedImageDirectoryName = "CHESS_DETECTED";
+        private const string DSLRArUcoDirectoryName = "DSLR_ARUCO";
+        private const string DSLRArUcoDetectedDirectoryName = "DSLR_ARUCO_DETECTED";
+        private const string HeadsetDataDirectory = "META";
+        private const string RootDirectoryName = "Calibration";
+
+        public static void Initialize(out int nextChessboardImageId, out int nextArUcoImageId)
+        {
+            nextChessboardImageId = 0;
+            nextArUcoImageId = 0;
+
+            string rootFolder = Path.Combine(GetDocumentsFolderPath(), RootDirectoryName);
+            if (Directory.Exists(rootFolder))
+            {
+                InitializeDirectory(rootFolder, ChessboardImageDirectoryName, "*.png", out nextChessboardImageId);
+                InitializeDirectory(rootFolder, DSLRArUcoDirectoryName, "*.png", out nextArUcoImageId);
+                InitializeDirectory(rootFolder, HeadsetDataDirectory, "*.json", out nextArUcoImageId);
+            }
+            else
+            {
+                Directory.CreateDirectory(rootFolder);
+                Directory.CreateDirectory(Path.Combine(rootFolder, ChessboardImageDirectoryName));
+                Directory.CreateDirectory(Path.Combine(rootFolder, ChessboardDetectedImageDirectoryName));
+                Directory.CreateDirectory(Path.Combine(rootFolder, DSLRArUcoDirectoryName));
+                Directory.CreateDirectory(Path.Combine(rootFolder, DSLRArUcoDetectedDirectoryName));
+                Directory.CreateDirectory(Path.Combine(rootFolder, HeadsetDataDirectory));
+            }
+
+            DirectoryInfo detectedChessboards = new DirectoryInfo(Path.Combine(rootFolder, ChessboardDetectedImageDirectoryName));
+            foreach (FileInfo file in detectedChessboards.GetFiles())
+            {
+                file.Delete();
+            }
+
+            DirectoryInfo detectedArUcos = new DirectoryInfo(Path.Combine(rootFolder, DSLRArUcoDetectedDirectoryName));
+            foreach (FileInfo file in detectedArUcos.GetFiles())
+            {
+                file.Delete();
+            }
+        }
+
+        private static void InitializeDirectory(string rootFolder, string directoryName, string fileExtensionExpected, out int nextImageId)
+        {
+            nextImageId = 0;
+            string directory = Path.Combine(rootFolder, directoryName);
+            if (Directory.Exists(directory))
+            {
+                nextImageId = Math.Max(Directory.GetFiles(directory, fileExtensionExpected, SearchOption.TopDirectoryOnly).Length, nextImageId);
```

Comment:
>Math.Max(Directory.GetFiles(directory, fileExtensionExpected, SearchOption.TopDirectoryOnly).Length, nextImageId); [](start = 30, length = 114)

This could be simplified to nextImageId = Directory.GetFiles(directory, fileExtensionExpected, SearchOption.TopDirectoryOnly).Length.

It's not possible for nextImageId to be anything greater than 0 in this case.

The main reason why I think it this should be simplified is that this code suggests that it's possible for nextImageId to be meaningful up until this point (that it could be > 0) - this is not the case